lib/cfn-nag/custom_rules/CognitoUserPoolMfaConfigurationOnorOptionalRule.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require_relative 'base'
+
+class CognitoUserPoolMfaConfigurationOnorOptionalRule < BaseRule
+  def rule_text
+    "AWS Cognito UserPool should have MfaConfiguration set to 'ON' (MUST be wrapped in quotes) or at least 'OPTIONAL'"
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F85'
+  end
+
+  def audit_impl(cfn_model)
+    violating_userpools = cfn_model.resources_by_type('AWS::Cognito::UserPool').select do |userpool|
+      violating_userpool?(userpool)
+    end
+
+    violating_userpools.map(&:logical_resource_id)
+  end
+
+  private
+
+  def yaml_reserved_keyword_boolean_conv?(user_pool, param_value = '')
+    if [true, false].include?(user_pool.mfaConfiguration)
+      true
+    elsif [true, false].include?(param_value)
+      true
+    end
+  end
+
+  def keyword_uppercase?(user_pool)
+    violating_key_words = %w[On on OFF Off off Optional optional]
```

Comment:
this code is replicated in a few places.... can't you just do a case insensitive request or force the case for comparison?

Reply:
So I was able to eliminate the code duplication for this, but I think I still need to catch every single case of those options (Specifically for the 'ON' and 'OPTIONAL' cases) because CloudFormation will only accept those values in full uppercase.